src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -944,8 +948,16 @@ public Void caseObjectType(ObjectType type) {
           } else if (type.isDict()) {
             emit("{[key: string]: any}");
           } else if (type.getReferenceName() != null) {
+            String name;
+            // If a class A is declared in goog.module and exported as default, closure
+            // assigns it a synthetic type $jscomp.scope.A.
+            if (type.getReferenceName().contains("$jscomp.scope")) {
```

Comment:
Add a comment on why/how/whether that is different for non-default exports? This seems a bit crazy, maybe it's a thing to raise with Closure? I'd expect to get wonky error messages for users, too (`cannot assign blah to type $jscomp.scope.A` - wtf?).